core/templates/dev/head/pages/exploration_editor/translation_tab/TranslationTabDirective.js

```diff
@@ -25,11 +25,17 @@ oppia.directive('translationTab', [
         '/pages/exploration_editor/translation_tab/' +
         'translation_tab_directive.html'),
       controller: [
-        '$scope', function($scope) {
-          $scope.$on('refreshTranslationTab', $scope.refreshTranslationTab);
-          $scope.refreshTranslationTab = function() {
-            // This will be used in future to intialize translation services.
-          };
+        '$scope', '$rootScope', 'ExplorationStatesService',
+        'EditorStateService', function(
+            $scope, $rootScope, ExplorationStatesService, EditorStateService) {
+          $rootScope.loadingMessage = 'Loading';
+          $scope.$on('refreshTranslationTab', function() {
+            if (ExplorationStatesService.getState(
+              EditorStateService.getActiveStateName())) {
+              $rootScope.$broadcast('refreshStateTranslation');
+            }
+          });
+          $rootScope.loadingMessage = '';
```

Comment:
Is this supposed to be inside the $scope.$on function? Right now it seems like it sets it to 'Loading' and then immediately sets it to blank, not sure though. Because $scope.$on is just technically setting a listener right? And then it should continue immediately?

Reply:
Done!